### DISCUSSION TITLE
How to fetch data from secured Endpoint in the getStatic-Methods?

### DISCUSSION THREAD

**Discussion by Maxinger15** [vercel/next.js#27860] [Help] [answered]
Hello,
In the getStaticProps and getStaticPaths Methods I want to load data  from a secured REST endpoint. 
For this I need an access token which I get over openid from Keycloak. 
Since this is at build time this has nothing to do with user depended data.

What is a good way to manage the token and make it accessable in several files in the getStaticProps and getStaticPaths Methods ? 
Since I didn't find anything about that problem, maybe I am totaly on the wrong way.

**Comment by leerob** [comment, marked as answer]
You can use [Environment Variables](https://nextjs.org/docs/basic-features/environment-variables) to securely store the access token to connect to your API. Since `getStaticProps` and other APIs run on the server (Node.js), you can safely use private keys to connect to external services. For example, you might forward this token as a `Bearer` token for the `Authorization` header.
